### DISCUSSION TITLE
watch account balance by default on `ConnectButton`

### DISCUSSION THREAD

**Discussion by mintdart** [rainbow-me/rainbowkit#626] [Ideas]
It'd be nice to have `watch` (https://wagmi.sh/docs/hooks/useBalance#watch-optional) set to true by default on `ConnectButton` component or atleast provide a prop `watchBalance`, right now I need to refresh page to see updated balance after a transaction
![Screen Shot](https://user-images.githubusercontent.com/96025197/181839456-4c058fd1-0592-4031-ac51-ed2a2809bd91.png)

**Comment by carletex** [comment]
I agree. Having a `watch` / `pollingTime` to update the balance will be nice.

**Comment by amoossssss** [comment]
Any updates on adding the `watch` props to `ConnectButton`?

**Comment by JavierFerrerGomez** [comment]
I ended up here trying to find out how to do this. It would be nice indeed :)

**Reply by amoossssss** [reply]
@JavierFerrerGomez 
What I end up doing is manually trigger a re-render on the ConnectButton component. 
For instance, update the key of the component.

**Comment by 0xmkh** [comment]
@mintdart The reason we didn't want to include `watch` is because it can be too expensive to call the RPC too many times or even have a timer to pull it every time. Instead we're working adding a hook that you can use to refetch the balance on RainbowKit whenever you complete a successful transaction. 

This way it'll be 10x easier and you won't have to watch your balance every few seconds or so, but whenever you have to when it's necessary.

Here is the PR for more details if you're interested: https://github.com/rainbow-me/rainbowkit/pull/1717
